calendar/experiments/calendar/parent/ext-calendar-provider.js

```diff
@@ -258,6 +258,34 @@ class ExtCalendarProvider extends cal.provider.BaseClass {
   }
 }
 
+class ExtFreeBusyProvider {
+  QueryInterface = ChromeUtils.generateQI(["calIFreeBusyProvider"]);
+
+  constructor(fire) {
+    this.fire = fire;
+  }
+
+  async getFreeBusyIntervals(aCalId, aRangeStart, aRangeEnd, aBusyTypes, aListener) {
+    try {
+      const TYPE_MAP = {
+        free: Ci.calIFreeBusyInterval.FREE,
+        busy: Ci.calIFreeBusyInterval.BUSY,
+        unavailable: Ci.calIFreeBusyInterval.BUSY_UNAVAILABLE,
+        tentative: Ci.calIFreeBusyInterval.BUSY_TENTATIVE,
+      };
+      let attendee = aCalId.replace(/^mailto:/, "");
+      let start = aRangeStart.icalString;
+      let end = aRangeEnd.icalString;
```

Comment:
For the record I think we might want to use rfc3339 format for dates, also in anticipation of the Temporal proposal. No need to change this here though, I know you used icalString for consistency with what I did in convertItem.

Reply:
I for one would prefer `2014-06-06T19:00:00.000Z` format.